types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -66,6 +93,60 @@ export interface APIGatewayProxyResult {
     isBase64Encoded?: boolean;
 }
 
+/**
+ * Works with HTTP API integration Payload Format version 2.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
+export interface APIGatewayProxyEventV2 {
+    routeKey: string;
+    rawPath: string;
+    rawQueryString: string;
+    cookies: string[];
+    headers: { [name: string]: string };
+    queryStringParameters: { [name: string]: string };
+    requestContext: {
+      accountId: string;
+      apiId: string;
+      authorizer: { jwt: {
+          claims: { [name: string]: string };
+          scopes: string[]
+          }
+      };
+      domainName: string;
+      domainPrefix: string;
+      http: {
+        method: string;
+        path: string;
+        protocol: string;
+        sourceIp: string;
+        userAgent: string;
+      };
+      requestId: string;
+      routeKey: string;
+      stage: string;
+      time: string;
+      timeEpoch: number;
+    };
```

Comment:
Isn't this `APIGatewayEventRequestContextWithAuthorizer<TAuthorizerContext>`? I don't think they changed anything in there....
Getting the authorizer stuff right is particularly messy, as you can probably tell from all the types...

Reply:
No. This is significantly different. For example, v1 has `identity` in `requestContext`, which doesn't exist in v2. Similarly, v2 has `http` in `requestContext` which doesn't exist in v1.